docs/bounding_volume_hierarchy.md

```diff
@@ -0,0 +1,188 @@
+# `ArborX::BVH`
+Defined in header [`<ArborX_LinearBVH.hpp>`](https://github.com/arborx/ArborX/blob/master/src/ArborX_LinearBVH.hpp)
+```C++
+template <typename DeviceType>
+class BoundingVolumeHierarchy;
+```
+
+The class template `ArborX::BoundingVolumeHierarchy` is a tree data structure that can be used to accelerate the search for geometrical objects in space.
+
+ArborX defines the `ArborX::BVH` alias template for convenience.
+```C++
+template <typename DeviceType>
+using BVH = BoundingVolumeHierarchy<DeviceType>;
+```
+
+## Template parameter
+`DeviceType`
+: A valid Kokkos device type, e.g. `Kokkos::Cuda::device_type` or `Kokkos::OpenMP::device_type`.
+
+## Member types
+Member type | Definition
+--- | ---
+`device_type` | `DeviceType`
+`size_type` | `DeviceType::memory_space::size_type`
+`bounding_volume_type` | `ArborX::Box`
+
+## Member functions
+[`(constructor)`](#arborxbvhdevicetypebvh)
+: constructs the tree data structure.
```

Comment:
I would prefer having a table here as well.

Reply:
Not gonna happen :p